src/serverrpc.cpp

```diff
@@ -81,18 +90,27 @@ CServerRpc::CServerRpc ( CServer* pServer, CRpcServer* pRpcServer, QObject* pare
             {
                 continue;
             }
+
             QJsonObject client{
                 { "id", i },
                 { "address", vecHostAddresses[i].toString ( CHostAddress::SM_IP_PORT ) },
                 { "name", vecsName[i] },
                 { "jitterBufferSize", veciJitBufNumFrames[i] },
                 { "channels", pServer->GetClientNumAudioChannels ( i ) },
+                { "instrumentCode", vecChanInfo[i].iInstrument },
+                { "instrumentName", CInstPictures::GetName ( vecChanInfo[i].iInstrument ) },
+                { "city", vecChanInfo[i].strCity },
+                { "countryCode", vecChanInfo[i].eCountry },
+                { "countryName", QLocale::countryToString ( vecChanInfo[i].eCountry ) },
+                { "skillLevelCode", vecChanInfo[i].eSkillLevel },
+                { "skillLevelName", SkillLevelToString ( vecChanInfo[i].eSkillLevel ) },
             };
             clients.append ( client );
         }
 
         // create result object
         QJsonObject result{
+            { "connections", pServer->GetNumberOfConnectedClients() },
```

Comment:
I believe vecHostAdddresses.Size() returns the number of slots, not the number of active connections.

Reply:
But I see this at line 86:
```
// fill list with connected clients
for ( int i = 0; i < iNumChannels; i++ )
```

Reply:
Oh and at line 60
```
/// @result {array} result.clients - The list of connected clients.
```

Reply:
Rather than call GetNumberOfConnectedClients, having actively filtered for the connected clients, just keep your own local count and use it.  You know you'll get the right number, that way.  Nothing will have changed between one call and the next.

Reply:
iNumChannels is the size of vecHostAddresses which 150 with the standard compile -- representing the max size of the all these vectors which is hard-coded as 15 (global.h:209)

It is true that I could count the loop through, but with large # of active connections would represent 'x' number of operations as opposed to a single call to a method that already knows the answer.

Reply:
i++ no more than 150 times is probably a _lot_ cheaper than the method call (allocate memory for the stack, put return address on stack, allocate memory for any returned values, make the call, copy local variable to stack, return from method, copy stack to local variable, free stack... depends how expensive getting the memory turns out to be...).

But imagine now you're in a separate thread.  And the JSONRPC code _should_ be running in a separate thread from the real-time server code.  You're now not only doing that in process call, you're doing a cross-thread context switch.